numpy/lib/format.py

```diff
@@ -163,6 +163,7 @@
 
 MAGIC_PREFIX = b'\x93NUMPY'
 MAGIC_LEN = len(MAGIC_PREFIX) + 2
+ARRAY_ALIGN = 64 # plausible values are powers of 2 between 16 and 4096
```

Comment:
Also curious as to why 4096 is the limit

Reply:
Letting the user pass in the alignment is not a bad enhancement idea, but that's probably a 0.1% use case.  64 bytes is a good minimum alignment not only due to AVX-512 but also because it is one cache line on x86.  There's virtually no benefit to less than 64, and as of 2017 virtually no motivation for more either.  Letting the user specify a larger value might be handy a few years from now.

4096 is the upper limit of "plausible" because it is one entire page on Linux, and mmap() can be offset at page boundaries on Linux, so on the platforms I know about, padding to more aligned than 4096 bytes would be useless (of course there must be some exceptional case lurking somewhere, but that's 0.001%).